src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/UserDefinedTypeDescriptor.cs

```diff
@@ -622,8 +621,11 @@ private uint GetClassTypeIndex(TypeDesc type, bool needsCompleteType)
                 }
                 else
                 {
-                    if (!hasInstanceFields)
-                        continue;
+                    // Allow instance fields to be emitted even for types that are never instantiated.
```

Comment:
Doing this is problematic because we're going to be looking at layout of types that we didn't look at before (no constructed type means we never had to calculate type size or look at instance fields). If there's a problem with the type (e.g. has an instance field of a type that doesn't resolve because e.g. NuGet was a mismatched version), we're likely to crash here a couple lines down. We had crashes like that before, this is what the code around line 636 is dealing with.

I'm going to ask about the motivation in the original issue. Right now I'm leaning toward By Design/Won't fix on this.